pandas/core/generic.py

```diff
@@ -12910,10 +12909,21 @@ def make_doc(name: str, ndim: int) -> str:
     else:
         raise NotImplementedError
 
+    if ndim == 1:
+        return_type = "Series or scalar"
+        axis_descr = "{index (0)}"
+        name2 = "Series"
+        if base_doc in (_num_doc, _sum_prod_doc):
+            return_type = "scalar"
+    else:
+        return_type = "Series or scalar"
+        axis_descr = "{index (0), columns (1)}"
+        name2 = "DataFrame"
+
```

Comment:
I don't think this logic is totally correct.

If I'm not missing anything, there are 3 cases:

- `Series`: return type is always `scalar`, the axis parameter only accepts 0 which is ignored.
- `DataFrame` group 1 (e.g. `skew`): axis accepts 0 and 1, return is always `Series`
- `DataFrame` group 2 (e.g. `any`): axis accepts 0, 1 and None, return is  `Series or scalar` since `axis=0` means both axis together and makes the return a scalar.

You'll have to check which base_doc belong to each group. Also, can you please rename `name2` to `obj_type` or something more descriptive?
Also, if you

Reply:
@datapythonista 

In the series case for:

https://pandas.pydata.org/docs/reference/api/pandas.Series.sem.html
https://pandas.pydata.org/docs/reference/api/pandas.Series.std.html
https://pandas.pydata.org/docs/reference/api/pandas.Series.var.html

In all of the above cases return type is Series or scalar as per documentation.

Reply:
I have made the suggested variable name, etc and modified logic for all cases, verified locally all the cases

## For Series

| Function | Official Documentation return type | PR code Return type |
|----------|-------|--------|
| Any      | Scalar or Series | Scalar or Series |
| All      | Scalar or Series | Scalar or Series |
| Min      | Scalar or scalar | Scalar |
| Max      | Scalar or scalar | Scalar |
| Sum      | Scalar or scalar | Scalar |
| Prod     | Scalar or scalar | Scalar |
| Median   | Scalar or scalar | Scalar |
| Mean     | Scalar or scalar | Scalar |
| Var      | Scalar or Series (if level specified) | Scalar or Series (if level specified) |
| Std      | Scalar or Series (if level specified) | Scalar or Series (if level specified) |
| Sem      | Scalar or Series (if level specified) | Scalar or Series (if level specified) |
| Skew     | Scalar or scalar | Scalar |
| Kurt     | Scalar or scalar | Scalar |
| Cumsum   | Scalar or Series | Scalar or Series |
| Cumprod  | Scalar or Series | Scalar or Series |
| Cummin   | Scalar or Series | Scalar or Series |
| Cummax   | Scalar or Series | Scalar or Series |


## For DataFrame


| Function | Official Documentation return type | PR code Return type |
|----------|-------|--------|
| Any      | Series or DataFrame | Series or DataFrame |
| All      | Series or DataFrame | Series or DataFrame |
| Min      | Series or scalar | Series or scalar |
| Max      | Series or scalar | Series or scalar |
| Sum      | Series or scalar | Series or scalar |
| Prod     | Series or scalar | Series or scalar |
| Median   | Series or scalar | Series or scalar |
| Mean     | Series or scalar | Series or scalar |
| Var      | Series or DataFrame (if level specified) | Series or DataFrame (if level specified) |
| Std      | Series or DataFrame (if level specified) | Series or DataFrame (if level specified) |
| Sem      | Series or DataFrame (if level specified) | Series or DataFrame (if level specified) |
| Skew     | Series or scalar | Series or scalar |
| Kurt     | Series or scalar | Series or scalar |
| Cumsum   | Series or DataFrame | Series or DataFrame |
| Cumprod  | Series or DataFrame | Series or DataFrame |
| Cummin   | Series or DataFrame | Series or DataFrame |
| Cummax   | Series or DataFrame | Series or DataFrame |

Reply:
Thanks for all the detailed information and for the work on this PR @YashpalAhlawat.

> In the series case for:
> 
> https://pandas.pydata.org/docs/reference/api/pandas.Series.sem.html https://pandas.pydata.org/docs/reference/api/pandas.Series.std.html https://pandas.pydata.org/docs/reference/api/pandas.Series.var.html
> 
> In all of the above cases return type is Series or scalar as per documentation.

Looking at for example `Series.std`, I fail to see a way for the function to return anything that it's not a scalar, the `level` argument was deprecated and doesn't exist anymore. So the documentation is outdated. I assume it's also the case for the other two functions.

For `Series.any` and `Series.all` I think a scalar is always returned as well.

Reply:
Also, for `DataFrame.any` and `DataFrame.all` the return time should be `Series or scalar`, not `Series or DataFrame`.

Reply:
@datapythonista ,

For series Std, any , all methods needs to updated. If level is deprecated from all std, var, sem.

For dataframe any,all methods need to be updated. Is level is deprecated from dataframe methods also.

Kindly, clear the expectations. I will fix it.

Reply:
You can check the docstring of the objects. For `var` for example, there is no parameter `levels`: https://pandas.pydata.org/docs/reference/api/pandas.Series.var.html

I think it's the case for all them, I checked several and I didn't see any with the `level` parameter.

Reply:
For series, all cases will return **Scalar**, except **std, var, sem** as all belongs to same group  should have **Series and Scalar** **_num_ddof_doc**
>>> df = pd.DataFrame({'a': [1, 2], 'b': [2, 3]}, index=['tiger', 'zebra'])
>>> type(df.sem())
<class 'pandas.core.series.Series'>

For DataFrame, all cases should return Series and Scalar, since **skew, max, min, ** belongs to same group **_num_doc**.

If this looks good to you i will proceed and implement.

@datapythonista please review

Reply:
For Series, cumsum and similar returns Series I think. And for Dataframe I think it can return DataFrame or Series as you had. For the rest what you say sounds correct. You can check the type annotations and examples, and check the parameter axis which is the one that controls what object is being returned.